assets/helpers/checkoutForm/checkoutForm.js

```diff
@@ -78,14 +79,48 @@ export const invalidReason = (form: Object | null): string => {
   }
 };
 
-
 export const formElementIsValid = (formElement: Object | null) => {
   if (formElement && formElement instanceof HTMLFormElement) {
     return formElement.checkValidity();
   }
   return false;
 };
 
+type FlowType = 'opf' | 'npf';
+
+export const onFormSubmit = (
+  flowPrefix: FlowType,
```

Comment:
with this many params, an object makes sense to give the params names at the call sites and make it more readable. you can actually destructure them into local variables right in the parameter list:

```javascript
const x = ({a, b, c}) => {console.log(a);}
```

Reply:
I'm not sure I agree entirely in this situation, as most of them are passed in already named very clearly, i,e `contributionType`, `paymentMethod` etc. But I take your point on the flow prefix. A good compromise would be to store all the parameters as variables before passing them through (which nearly all of them already are anyway), that way you don't have to add loads of boilerplate at the other end. What do you think?

Reply:
then, to use ContributionForm as an example, you call it with something like this: 

`
   const flowPrefix = 'npf';
    const form = event.target;
    const handlePayment = () => formHandlers[props.contributionType][props.paymentMethod](props);
    onFormSubmit(
      flowPrefix,
      props.paymentMethod,
      props.contributionType,
      form,
      props.isSignedIn,
      props.userTypeFromIdentityResponse,
      props.setFormIsValid,
      props.setCheckoutFormHasBeenSubmitted,
      handlePayment,
    );
`

Reply:
actually, I agree with you more looking at how it's called in `ContributionsLandingActions`. I'll give it a go and see how it feels.

Reply:
but then don't you have boilerplate (i.e. storing them in variables) at every call site? vs boilerplate once in the function definition?

I take the point in general about the variable names but I flagged an example in `assets/pages/regular-contributions/components/contributionsGuestCheckoutContainer.jsx` where I think param names would be helpful (e.g., what are we passing an empty function for?)

Reply:
You were right, it's much nicer now. Have a look and see what you think